shared_task_code/task_queue_consumer.py

```diff
@@ -56,8 +56,12 @@ def start(self, ValidateData=None, TaskCode=None):
             scan = loads(record["body"])
             message_id = f"{record['messageId']}"
             if self.func_validatedata != None:
-                valid = self.func_validatedata(self.event, scan, message_id)
+                (valid, scan) = self.func_validatedata(self.event, scan, message_id)
             else:
                 valid = True
             if valid:
-                self.run_scan(scan, message_id)
+                if isinstance(scan, list):

```

Comment:
This duplicated code should be avoided. ECSTaskQueueConsumer should be extending a base TaskQueueConsumer which already does all of this.

Reply:
now extends the task consumer